File: academic-system-java/src/main/java/org/example/academic/system/model/Exam.java
package org.example.academic.system.model;

public class Exam extends Assessment {
    // Cria uma avaliação do tipo Exame com valor e peso informados
    public Exam(double value, double weight) {
        super(value, weight);
    }

    // Retorna o identificador do tipo desta avaliação
    @Override
    public String getType() {
        return "Exam";
    }
}

File: academic-system-python/academic_system/model/exam.py
from .assessment import Assessment


class Exam(Assessment):
    # Cria uma avaliação do tipo Exame com valor e peso informados
    def __init__(self, value: float, weight: float):
        super().__init__(value, weight)

    # Retorna o identificador do tipo desta avaliação
    @property
    def type(self) -> str:
        return "Exam"
